stdlib/Logging/src/ConsoleLogger.jl

```diff
@@ -96,38 +96,41 @@ function termlength(str)
 end
 
 function handle_message(logger::ConsoleLogger, level, message, _module, group, id,
-                        filepath, line; maxlog=nothing, kwargs...)
```

Comment:
Why remove `maxlog=nothing` from here? It seems to make the code below rather more complicated.

Reply:
Saves the compiler/frontend from needing to create a keyword sorter and corresponding structdiff methods

Reply:
I don't understand why this is worthwhile... unless something about this API forces the amount of compiler work to scale with the number of logging statements in the program?

Is the point that `kwargs` can have many different types (one per logging statement) and that leads to trouble?

If so that connects to another idea I've been thinking about: That perhaps we should be using a data structure more like `Dict{Symbol,Any}`, or `ImmutableDict{Symbol,Any}` for passing the key-value pairs of log records rather than (ab)using keyword arguments?

Reply:
With this change, kwargs can already be (already is) an AbstractDict{Symbol,<:Any}, without making any further changes to the code